modules/hostap/Kconfig

```diff
@@ -175,6 +175,10 @@ config WIFI_NM_WPA_SUPPLICANT_BSS_MAX_IDLE_TIME
 config WIFI_NM_WPA_SUPPLICANT_NO_DEBUG
 	bool "Disable printing of debug messages, saves code size significantly"
 
+config WIFI_NM_WPA_SUPPLICANT_INF_MON
+	bool "Monitor the net mgmt event to add/del interface"
+	default y
```

Comment:
Why do we need this, I mean when would we want to turn this interface monitoring off?

Reply:
And without a interface added to the WPA supplicant it's useless.

Reply:
Hi @jukkar @krish2718 , actually we turn this monitoring off in our chip and currently we don't need to delete interface when the net_if down. 
the setup_interface_monitoring() already call net_if_foreach(iface_cb, ctx), and iface_cb will add_interface.

Reply:
Yes, currently we only add interface at the init time and the interface is always present.
add or del interface based on the netif up/down is not used by us, I think add this macro is more clean and user can choose enable it or disable it. Currently disable it is more easy for us to develop upstream supplicant.